atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPoolImpl.java

```diff
@@ -346,6 +347,29 @@ void setServersInPoolTo(ImmutableMap<CassandraServer, CassandraServerOrigin> des
                 .removeKeys(currentServers::contains)
                 .toImmutableMap();
         SetView<CassandraServer> absentServers = Sets.difference(currentServers, desiredServers.keySet());
+        Map<CassandraServer, CassandraClientPoolingContainer> currentPoolWithoutAbsentServers = EntryStream.of(
+                        getCurrentPools())
+                .removeKeys(absentServers::contains)
+                .toMap();
+
+        Set<CassandraServer> validatedServersToAdd =
+                validateNewHostsTopologiesAndMaybeAddToPool(currentPoolWithoutAbsentServers, serversToAdd);
+
+        boolean poolEmpty = currentPoolWithoutAbsentServers.isEmpty() && validatedServersToAdd.isEmpty();
+
+        if (poolEmpty) {
+            metrics.recordEmptyPool();
```

Comment:
ok a bit uggs, but my rationale here is that it looks even worse with `Preconditions.checkState`, plus we already have the `if` statement for the metrics, so might as well.